app/cdap/components/DataPrep/TopPanel/PipelineConfigHelper.js

```diff
@@ -53,20 +53,22 @@ export default function getPipelineConfig() {
 
     res.sources.forEach((plugin) => {
       const pluginType = plugin.plugin.type;
-
-      const connections = pluginType === 'batchsource' ? batchConnections : realtimeConnections;
-      const stages = pluginType === 'batchsource' ? batchStages : realtimeStages;
-
-      const pluginStage = formatPluginSpec(plugin);
-      const pluginName = pluginStage.name;
-
-      const connectionObj = {
-        from: pluginName,
-        to: wranglerName,
-      };
-
-      connections.push(connectionObj);
-      stages.push(pluginStage);
+      // add if to prevent making batchsink into realtime source
+      if (pluginType !== 'batchsink') {
```

Comment:
nitpick alternatively, we could return early when we found the condition check is false:
`
if (pluginType === 'batchsink') {
  return;
}
`

Reply:
better change this to 
```
if (pluginType !== 'batchsource' && pluginType !== 'streamingsource') {
  return;
}
```
The only two plugin type that can be sources in the pipelines are batchsource and streamingsource, so later your logic should be:
1. If this is a batchsource, allow batch pipeline creation
2. If this is a streamingsource, allow streaming pipeline creation
3. If both types are included, allow batch and streaming pipeline creation